packages/fuels-signers/src/provider.rs

```diff
@@ -67,10 +64,26 @@ impl Provider {
     ///   Ok(())
     /// }
     /// ```
-    pub async fn send_transaction(&self, tx: &Transaction) -> io::Result<Vec<Receipt>> {
-        let tx_id = self.client.submit(tx).await?;
+    pub async fn send_transaction(&self, tx: &Transaction) -> Result<Vec<Receipt>, Error> {
+        let (status, receipts) = self.submit_with_feedback(tx).await?;
+
+        match status {
+            TransactionStatus::Failure { reason, .. } => {
+                Err(Error::RevertTransactionError(reason, receipts))
+            }
+            _ => Ok(receipts),
+        }
```

Comment:
```suggestion
        if let TransactionStatus::Failure { reason, .. } = status {
            Err(Error::RevertTransactionError(reason, receipts))
        } else {
            Ok(receipts)
        }
```

Looks simpler?

Reply:
Potato potato 😁

Reply:
No u didn't :D It has one indent less! A universal banana measurement.